test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -325,11 +333,13 @@ public boolean shouldCache(Query query) {
         }
         when(ctx.fetchPhase()).thenReturn(new FetchPhase(Arrays.asList(new FetchSourcePhase(), new FetchDocValuesPhase())));
         when(ctx.getQueryShardContext()).thenReturn(queryShardContext);
-        NestedDocuments nestedDocuments = new NestedDocuments(mapperService, bitsetFilterCache::getBitSetProducer);
-        when(ctx.getNestedDocuments()).thenReturn(nestedDocuments);
         IndexShard indexShard = mock(IndexShard.class);
         when(indexShard.shardId()).thenReturn(new ShardId("test", "test", 0));
         when(ctx.indexShard()).thenReturn(indexShard);
+        MapperService mapperService = mock(MapperService.class);
+        when(mapperService.hasNested()).thenReturn(false);
```

Comment:
This one throws me off: why do we need to mock hasNested now compared to before? It looks to me that these test changes are not strictly required and could be made as a follow-up. But I am probably missing something.

Reply:
This is because the agg tests never want to try and fetch nested hits but are quite happy to do the aggs on them. Its *weird* but this funny dance keeps them working the way they were before.

Reply:
Previously we bumped into this behavior because of the defaults of mockito - it returns `false` unless you tell it otherwise. So I'm making this explicit....

Reply:
I see makes sense